### PR TITLE
Update with latest changes to LPI2 spec

### PR DESCRIPTION
**Pull request by justmoon**

### CODE REVIEW

File: errors/InterledgerRejectionError.js [hunk 1]
```diff
@@ -2,46 +2,12 @@
...
-    this.ilpRejection = {
-      code,
-      name: opts.name || CODE_NAMES[code] || 'Unknown',
-      message,
-      triggeredBy: opts.triggeredBy || '',
-      forwardedBy: opts.forwardedBy || [],
-      triggeredAt: opts.triggeredAt || new Date(),
-      additionalInfo: opts.additionalInfo || { message }
-    }
+    this.ilpRejection = opts.ilpRejection
```
Comment (emschwartz) [inline errors/InterledgerRejectionError.js:10]: Is it weird that this can be an `IlpRejectionPacket` or an `IlpErrorPacket`? Or is the thought that the latter is going to be deprecated anyway?

File: index.js [hunk 2]
```diff
@@ -19,6 +20,8 @@ const PASSTHROUGH_EVENTS = [
   'info_change'
 ]
 
+const COMPAT_SYMBOL = Symbol.for('org.interledgerjs.ilp-compat-plugin.idempotence')
```
Comment (emschwartz) [inline index.js:23]: What is this for?

**Review by emschwartz — APPROVED** [review]
We should add tests to this module, but these changes LGTM for now

File: errors/InterledgerRejectionError.js (hunk 1 above)
Comment (justmoon) [reply errors/InterledgerRejectionError.js:10]: https://gist.github.com/justmoon/7612998c5afdfa356c3f687839029168

TL;DR: When two LPI1 plugins are talking to each other, it'll be an ILP error and if it's two LPI2 plugins talking to each other it'll be an ILP rejection. If you're an LPI2 connector and you want to connect to an LPI1 peer, you need to use an LPI1 plugin wrapped in ilp-compat-plugin.

File: index.js (hunk 2 above)
Comment (justmoon) [reply index.js:23]: If someone passes the same plugin to compat twice, we want to return the same wrapper instance. The best way to do that is to add a property on the wrapped plugin that we can look at. The best way to ensure that that property doesn't collide with anything else is to use a Symbol.
